server/src/web/auth.controller.spec.ts

```
@@ -0,0 +1,118 @@
+
+      expect(await controller.createToken(createTokenRequestDto)).toBe(jwt);
+    });
+
+    it('should return undefined', async () => {
+      when(mockAuthService.createToken(address)).thenReturn(undefined);
+
+      controller = new AuthController(instance(mockAuthService));
+
+      expect(await controller.createToken(createTokenRequestDto)).toEqual(undefined);
+    });
+  });
+
+  describe('#signUp()', () => {
+    it('should create a user', async () => {
+      const jwt: Token = {
+        expiresIn: 3600,
+        accessToken: 'token',
+      } as Token;
+
+      const authDto = new SignUpResponseDto(jwt.accessToken, message, name);
+      when(mockAuthService.signUp(anyString(), anyString(), anyString(), anyString())).thenReturn(new Promise(resolve => {
+        resolve(authDto);
+      }));
+
+      controller = new AuthController(instance(mockAuthService));
+
+      expect(await controller.signUp(signUpRequestDto)).toBe(authDto);
+    });
+
+    it('should return null', async () => {
+      when(mockAuthService.signUp(address, name, message, signature))
+        .thenReject().thenThrow(new UnauthorizedException());
+
+      controller = new AuthController(instance(mockAuthService));
+
+      expect(await controller.createToken(createTokenRequestDto))
+        .toEqual(undefined);
+    });
+  });
+
+  describe('#signIn()', () => {
+    it('should return a user', async () => {
+      const jwt: Token = {
+        expiresIn: 3600,
+        accessToken: 'token',
+      } as Token;
+      const authDto = new SignInResponseDto(jwt.accessToken, message, name);
+      when(mockAuthService.signIn(address, message, signature)).thenReturn(new Promise(resolve => {
+        resolve(authDto);
+      }));
+
+      controller = new AuthController(instance(mockAuthService));
+
+      expect(await controller.signIn(signInRequestDto)).toBe(authDto);
+    });
+
+    it('should throw "user not found"', async () => {
```

it says should throw error but returns undefined at the code